### ISSUE TITLE
ES6 Map object reported as not defined

### ISSUE THREAD

**Issue by MidnightLightning** [eslint/eslint#5674]
**What version of ESLint are you using?**
v2.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
espree

**Please show your full configuration:**
`eslint --print-config app.js` results:
```
{
  "globals": {},
  "env": {},
  "rules": {
    "no-undef": 2
  },
  "parserOptions": {
    "ecmaVersion": 6
  },
  "parser": "espree"
}
```
**What did you do? Please include the actual source code causing the issue.**
`app.js` contents:
```
'use strict';

let gauges = new Map();
```

**What did you expect to happen?**
That the `no-undef` rule when run with the `ecmaVersion:6` parserOption set would know that [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) is a new type, and would not report an error.

**What actually happened? Please include the actual, raw output from ESLint.**
`eslint app.js` reports the error:
```
3:18  error  'Map' is not defined  no-undef
```

**Comment by mysticatea** [comment]
Thank you for this issue.

Please use `es6` environment.

```js
{
    "env": {
        "es6": true
    }
}
```

`parserOptions` is options for syntax.
`env` defines global variables.

**Comment by damianobarbati** [comment]
Not working!

```
let taFuncs = new Set(); //throws no-undef
```

Package.json:
```
"eslintConfig": {
        "parser": "babel-eslint",
        "env": {
            "browser": true,
            "node": true,
            "jest": true
        },
        "parserOptions": {
            "ecmaVersion": 6,
            "sourceType": "module",
            "impliedStrict": true,
            "ecmaFeatures": {
                "jsx": true,
                "experimentalObjectRestSpread": true
            }
        },
        "plugins": [
            "import",
            "react",
            "jest"
        ],
        "extends": [
            "eslint:recommended",
            "plugin:react/recommended",
            "plugin:jest/recommended"
        ],
        "rules": {
            "no-console": "off",
            "no-unused-vars": "off",
            "no-unsafe-finally": "off",
            "react/no-unescaped-entities": "off",
            "react/prop-types": "off"
        }
    },
```
